Fixes for CRAN

jefferis opened the pull request:
* Some minor fixes that should make the package ready for CRAN again
* main potential issue is if there is still any fallout left from the rust test compilation in #69

jefferis committed:
Update to roxygen2 v6.0.1

jefferis committed:
Make GNU make a system requirement

* Required for `ifeq ($(shell ...` in Makevars

jefferis committed:
Makevars: FLIBS must be included if BLAS_LIBS present

* MAKE Check was complaining
* see  https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Using-Makevars

jefferis committed:
.onLoad replacing httr:::path.length no longer required

* r-lib/httr/issues/395 was closed in Jul 2017
* MAKE Check doesn't like this

jefferis committed:
correctly ignore neokit and figure directories

* "figure/" ignores everything underneath the directory but leaves the (empty) directory itself
* gave a message during CHECK

PlasmaPower commented:
Is the bolt fallback building properly on your system?

Also, I think we're using a lot of deprecated neo4j endpoints. Should we get rid of them before publishing to CRAN?

jefferis committed:
remove date field

* it's added on build

jefferis committed:
quote names in Description

* to keep check happy

jefferis committed:
Add registration of native routines

* another check issue revealed on winbuilder

jefferis commented:
bolt fallback seems to be OK now, but there are still some more issues to fix for CRAN. The main one is that there is a method clash with a recommended package:
```
** checking use of S3 registration ... WARNING
Registered S3 method from a standard package overwritten by 'RNeo4j':
  method     from
```
this looks like a difficult one to get round without changing the `path` class inside RNeo4j (it probably wasn't a good choice of class name in the beginning).

jefferis committed:
Rename to path class to neopath to avoid clashes with base packages

* specifically print.path

** checking use of S3 registration ... WARNING
Registered S3 method from a standard package overwritten by 'RNeo4j':
  method     from
  print.path grid

jefferis committed:
More fixes for path->neopath revealed by tests

jefferis committed:
rust/bolt paths also needs to inherit from neopath

jefferis commented:
This PR looks like it might pass all CRAN checks. It would be a good idea if anyone interested tried it out:

```
devtools::install_github("jefferis/RNeo4j@fix/CRAN")
```

SiggiSmara commented:
Just installed on a Linux Mint 18 and R 3.5.0 with no prior installments of RNeo4j, no problems installing. Tested startGraph(),  cyper() and cypherToList().  All worked fine.

Ran the following demo code from here, again no errors and the expected output.

``` r
nicole <- createNode(graph, "Person", name="Nicole", age=24)
greta <- createNode(graph, "Person", name="Greta", age=24)
kenny <- createNode(graph, "Person", name="Kenny", age=27)
shannon <- createNode(graph, "Person", name="Shannon", age=23)

r1 <- createRel(greta, "LIKES", nicole, weight=7)
r2 <- createRel(nicole, "LIKES", kenny, weight=1)
r3 <- createRel(kenny, "LIKES", shannon, weight=3)
r4 <- createRel(nicole, "LIKES", shannon, weight=5)

query <- "
MATCH (nicole:Person)-[r:LIKES]->(p:Person)
WHERE nicole.name = 'Nicole'
RETURN nicole.name, r.weight, p.name
"
cypher(graph, query)

query <- "
MATCH (nicole:Person)-[:LIKES]->(p:Person)
WHERE nicole.name = 'Nicole'
RETURN nicole, COLLECT(p.name) AS friends
"
cypherToList(graph, query)
```

jefferis committed:
upgrade to roxygen2 v6.1.1

jefferis commented:
@PlasmaPower I think you now have write access on this repo. Do you want to merge this PR? It still checks correctly for me – just one NOTE that gnu make is a system requirement.

jefferis commented:
Sample r-hub build log for r-release
https://builder.r-hub.io/status/RNeo4j_1.7.0.tar.gz-f08ca26e87cd40cf9dbe5ac4c334a1f4

jefferis commented:
Note that the check failure above is due to a pre-install neo4j build error on 1 / 4 parts of the build matrix, not to a problem with this package itself.
